src/js/core/directives/ui-grid-header-cell.js

```diff
@@ -328,6 +333,14 @@
 
             updateHeaderOptions();
 
+            if ($scope.col.filterContainer === 'columnMenu' && $scope.col.filterable) {
+              $rootScope.$on('menu-shown', function() {
```

Comment:
Why is $rootScope needed here?

Reply:
Also, it might make sense to have tests that specifically change that setFilter works as expected when menu-shown event is triggered,

Reply:
The `'menu-shown'` event is emitted from the `uiGridMenu` directive, which is outside the `uiGridHeaderCell` directive, so $rootScope is needed to listen for the event from within `uiGridHeaderCell`. I'm open to suggestions if you think there might be a better way. We could perhaps move the filter stuff to another directive (`'uiGrid'`?).